packages/sync-actions/src/zones-actions.js

```diff
@@ -25,20 +25,45 @@ export function actionsMapLocations(diff, oldObj, newObj) {
       action: 'addLocation',
       location: newLocation,
     }),
-    [REMOVE_ACTIONS]: oldLocation => ({
-      action: 'removeLocation',
-      location: oldLocation,
-    }),
-    [CHANGE_ACTIONS]: (oldLocation, newLocation) => [
-      {
-        action: 'removeLocation',
-        location: oldLocation,
-      },
-      {
-        action: 'addLocation',
-        location: newLocation,
-      },
-    ],
+    [REMOVE_ACTIONS]: oldLocation =>
+      // We only add the action if the location is not included in the new object.
+      !newObj.locations.find(
+        location => location.country === oldLocation.country
+      )
+        ? {
+            action: 'removeLocation',
+            location: oldLocation,
+          }
+        : null,
+    [CHANGE_ACTIONS]: (oldLocation, newLocation) => {
+      const result = []
+
+      // We only remove the location in case that the oldLocation is not
+      // included in the new object
+      if (
+        !newObj.locations.find(
+          location => location.country === oldLocation.country
+        )
+      )
+        result.push({
+          action: 'removeLocation',
+          location: oldLocation,
+        })
+
+      // We only add the location in case that the newLocation was not
+      // included in the old object
+      if (
+        !oldObj.locations.find(
```

Comment:
Having the same thing three times I wonder if we can extract it into a

```js
const hasLocation(locations, otherLocation) =>
   locations.some(location => location.country === otherLocation.country
```

and just use it everywhere like

```js
if(!hasLocation(oldObj.locations, newLocation) {}
```

Reply:
Uh! Good point 👍

Reply:
yeah, as I mentioned in my previous comment

Reply:
a180141 😎